pyvista/core/grid.py

```diff
@@ -652,6 +652,10 @@ def points(self) -> NumpyArray[float]:
                [1., 1., 1.]])
 
         """
+        # Handle empty case
+        if not all(self.dimensions):
+            return np.empty((0, 3))
```

Comment:
Do we also need to initialize this to zero?
```suggestion
            return np.zeros((0, 3))
```

Reply:
Seems like they produce the same result? Both are arrays are empty. Since the dimensions are `(0,3)` there are no actual values in the output
``` python
import numpy as np; np.zeros((0,3))
# Out[3]: array([], shape=(0, 3), dtype=float64)
import numpy as np; np.empty((0,3))
# Out[4]: array([], shape=(0, 3), dtype=float64)
```

But maybe it's better to change it to `np.zeros` to match the test

Reply:
Ah! If the size is zero, it's the same. I'll change it to make it same with test code. Thanks for the comment.

Reply:
I missed that it was used twice.  The fact that `np.empty` returns zeros is not guaranteed.  It _could_ return nonzeros on some systems.  Thanks for fixing.